driver/amc_pci_core.c

```diff
@@ -125,19 +138,39 @@ static int amc_pci_open(struct inode *inode, struct file *file)
     /* Replace the file's f_ops with our own and perform any device specific
      * initialisation. */
     int minor_index = iminor(inode) - amc_priv->minor;
-    file->f_op = fops_info[minor_index].fops;
+
     int rc = -EINVAL;
-    switch (minor_index)
+    struct prom_entry *pentry = prom_find_entry(amc_priv->prom, minor_index);
+
+    if (pentry)
     {
-        case MINOR_REG:
-            rc = amc_pci_reg_open(file, amc_priv->dev, amc_priv->interrupts, &amc_priv->locking);
-            break;
-        case MINOR_DDR0:
-            rc = amc_pci_dma_open(file, amc_priv->dma, DDR0_BASE, DDR0_LENGTH);
+        switch (pentry->tag)
+        {
+        case PROM_DEVICE_TAG:
```

Comment:
Interesting question.  Do we indent the case to the switch or one level deeper?  Check the rest of the code base, need to be consistent about this.

I always indent deeper:
```c
switch (blah)
{
    case label:
        body;
        break;
    case label2:
    {
        body with braces;
        break;
    }
}
```

Reply:
done